Add hide event and fixed some logics and bugs

Vaerrwenn opened the pull request:
Bottom text

Vaerrwenn committed:
Add types and categories tables and trying not to feck up this time for real.

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Add url to types and ctgs tbls

Vaerrwenn committed:
Images

Vaerrwenn committed:
Update types in DatabaseSeeder

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Type model, controller, and api

Vaerrwenn committed:
Update 2019_09_15_070809_create_users_table.php

Vaerrwenn committed:
Add tickets and transacions tables

yeet

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
remove camel case

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Added a lot of stuff

Event and ticket

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Create reports table and edited users, events, tickets, and comments table.

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Add image upload, search, and update event

Vaerrwenn committed:
Merge branch 'master' into servant

Vaerrwenn committed:
Hide events and some bugfixes

gavinclive commented on app/Event.php:
```diff
@@ -83,4 +84,12 @@ public function updateEventById($eventArr, $id)
+    #region Hide Event
+    public function hideEvent()
+    {
+        return $this->where('published_date', '>=', Carbon::now())
```
shouldn't this be `<=`, because we can only see events after they are published, not before they are published

gavinclive commented on database/seeds/DatabaseSeeder.php:
```diff
@@ -29,7 +29,7 @@ public function run()
             'email' => 'ucok@es.com',
             'password' => bcrypt('ucok'),
-            'role' => 2,
+            'role' => 3,
```
set ucok's role to `1` and eo's role to `3`

Vaerrwenn committed:
Fixed some things

When I took a shower, my mind got opened by the light of enlightenment. Now I understand the logic and I have proceed the logic.
